lib/osut/utils.rb

```diff
@@ -6274,133 +6667,112 @@ def addSkyLights(spaces = [], opts = {})
         bm2 = bm2.get
         next if bm2.round(2) < w02.round(2)
 
-        # Track if bounded box is significantly smaller than roof.
+        width = alignedWidth(box, true)
+        depth = alignedHeight(box, true)
+        next if width < wl * 3
+        next if depth < wl
+
+        # A set is 'tight' if the area of its bounded box is significantly
+        # smaller than that of its roof. A set is 'thin' if the depth of its
+        # bounded box is (too) narrow. If either is true, some geometry rules
+        # may be relaxed to maximize allocated skylight area. Neither apply to
+        # cases with skylight wells.
         tight = bm2 < roof.grossArea / 2 ? true : false
+        thin  = depth.round(2) < (1.5 * wl).round(2) ? true : false
 
         set           = {}
         set[:box    ] = box
         set[:bm2    ] = bm2
         set[:tight  ] = tight
+        set[:thin   ] = thin
         set[:roof   ] = roof
         set[:space  ] = space
+        set[:m      ] = space.multiplier
```

Comment:
A number of necessary changes have been introduced to correct the use of space multipliers (and elsewhere existing subsurface multipliers). Unfortunately no RSpec tests yet ... @todo.